netty-utils/src/test/java/org/asynchttpclient/netty/util/ByteBufUtilsTests.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2019 AsyncHttpClient Project. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ *     http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.asynchttpclient.netty.util;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import java.nio.charset.Charset;
+import org.testng.annotations.Test;
+import org.testng.Assert;
+import org.testng.internal.junit.ArrayAsserts;
+
+public class ByteBufUtilsTests {
+
+    @Test
+    public void testByteBuf2BytesEmptyByteBuf() {
+        ByteBuf buf = Unpooled.buffer();
+        ArrayAsserts.assertArrayEquals(new byte[]{},
+                ByteBufUtils.byteBuf2Bytes(buf));
+        buf.release();
```

Comment:
nit: could you please perform the release calls in a finally block so the ByteBufs get released even if the assert fails?

Reply:
Finally block added to test method